File: elixir/FormatCourses.exs
defmodule CourseFormatter do
  def format_course(course) do
    %{
      client: Map.get(course, "client"),
      id: Map.get(course, "course_id"),
      course: %{name: Map.get(course, "course_name")},
      environment: %{name: Map.get(course, "environment_name")},
      enrolled: Map.get(course, "enrolled"),
      accessed: Map.get(course, "accessed"),
      not_accessed: Map.get(course, "notAccessed"),
      percent_accessed: Map.get(course, "percentAccessed"),
      percent_not_accessed: Map.get(course, "percentNotAccessed")
    }
  end

  def format_courses(courses, page_params) do
    formatted_courses = Enum.map(courses, &format_course/1)
    %{
      courses: formatted_courses,
      item_count: length(formatted_courses),
      total_count: length(formatted_courses),
      page_params: page_params
    }
  end
end

File: java/FormatCourses.java
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

class FormatCourse {
    public static Map<String, Object> formatCourse(Map<String, Object> course) {
        return Map.of(
                "client", course.get("client"),
                "id", course.get("course_id"),
                "course", Map.of("name", course.get("course_name")),
                "environment", Map.of("name", course.get("environment_name")),
                "enrolled", course.get("enrolled"),
                "accessed", course.get("accessed"),
                "notAccessed", course.get("notAccessed"),
                "percentAccessed", course.get("percentAccessed"),
                "percentNotAccessed", course.get("percentNotAccessed")
        );
    }
    
    public static Map<String, Object> formatCourses(List<Map<String, Object>> courses, Map<String, Integer> pageParams) {
        List<Map<String, Object>> formattedCourses = courses.stream()
                .map(FormatCourse::formatCourse)
                .collect(Collectors.toList());
    
        return Map.of(
                "courses", formattedCourses,
                "itemCount", formattedCourses.size(),
                "totalCount", formattedCourses.size(),
                "pageParams", pageParams
        );
    }
}